lib/effects/combined_effect.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:math';
+
+import 'package:flame/components/component.dart';
+import 'package:meta/meta.dart';
+
+import './effects.dart';
+
+class CombinedEffect extends PositionComponentEffect {
+  final List<PositionComponentEffect> effects;
+  final double offset;
+
+  CombinedEffect({
+    @required this.effects,
+    this.offset = 0.0,
+    bool isInfinite = false,
+    bool isAlternating = false,
+  }) : super(isInfinite, isAlternating) {
+    final types = effects.map((e) => e.runtimeType);
+    // All effect types have to be different so that they don't clash
+    assert(types.toSet().length == types.length);
```

Comment:
Add a meaningful message here, could even be that comment above

Reply:
Good idea, I added it to SequenceEffect too (a bit dirty to do it in this PR, but I will totally forget it otherwise).